doc/fpmsyncd/frr_sonic_protobuf_channel.md

```diff
@@ -77,29 +114,39 @@ This feature requires:
 
 The following figure shows the changes to the SONiC Architecture:
 
 ![FRR-SONiC Protobuf Communication Channel - High-Level Architecture](images/fpm-pb-channel-architecture.png)	
-*Figure 2: FRR-SONiC Protobuf Communication Channel - High-Level Architecture*
+*Figure 3: FRR-SONiC Protobuf Communication Channel - High-Level Architecture*
 
 The new modules are colored in <span style="color:blue">blue</span>; the modified components are colored in <span style="color:green">green</span>.
 
 ## 7. High-Level Design 
 
 ### 7.1. FRR changes
 
-FRR has a module called `dplane_fpm_nl` which is responsible for exporting the routing information to SONiC. The routing information exported by `dplane_fpm_nl` is encoded using Netlink as the message format.
+In order to support transmitting information encoded in Protobuf format, we make some changes in FRR.
+
+First, we add a new component called `dplane_fpm` (referred to as FPM) in FRR.
+FPM establishes a communication channel with SONiC (specifically, SONiC `fpmsyncd`). This channel is used to transmit both Netlink and Protobuf messages.
+
+FPM exports an `fpm_encoder_register()` API, which is used to add FPM encoders into the FPM system.
+
+* We introduce a new module: the FPM Protobuf module (`dplane_fpm_pb`). This module provides capability to encode/decode information in Protobuf format. During its initialization, it uses the `fpm_encoder_register()` API to register as an FPM encoder.
+
+* We make some changes to the existing FPM Netlink module (`dplane_fpm_nl`). This module provides capability to encode/decode information in Netlink format. During its initialization, it uses the `fpm_encoder_register()` API to register as an FPM encoder.
 
-In order to support transmitting information encoded in Protobuf format, we introduce a new module in FRR. The new module is called `dplane_fpm_pb` and is responsible for encoding and sending information to SONiC using Protobuf as the message format. The new FPM module (`dplane_fpm_pb`) is designed to coexist with the existing one (`dplane_fpm_nl`). The two modules can be run simultaneously.
+* The FPM Protobuf module has higher priority than the FPM Netlink module.
 
+When FRR needs to transmit information, such as a route, to SONiC, it passes the data to FPM. Subsequently, FPM scans through the list of FPM encoders and selects the first encoder capable of encoding the requested data. Given that the Protobuf encoder holds a higher priority than the Netlink encoder, this means that FPM attempts to encode the information using the Protobuf encoder. If Protobuf is unable to encode the requested data, FPM falls back to the Netlink encoder. Following this, the chosen encoder encodes the information, and finally, FPM sends the resulting message to SONiC.
```

Comment:
does this mean when FPM protobuf is enabled then FRR might not use the netlink encoder for any of the information that needs to syncd to SONiC

Reply:
Ditto. In the updated HLD, we are no longer using Protobuf.